src/rtcModule/webrtc.cpp

```diff
@@ -853,6 +906,51 @@ void RtcModule::abortCallRetry(Id chatid)
     }
 }
 
+void RtcModule::refreshTurnServerIp()
+{
+    if (mIceServerProvider.empty())
+    {
+        return;
+    }
+
+    std::string url = mIceServerProvider[0]->url;
+    if (url.size())
+    {
+        size_t postInitialColon = url.find(":") + 1;
+        size_t postFinalColon = url.rfind(":");
+        std::string subString = url.substr(postInitialColon, postFinalColon - postInitialColon);
+        auto wptr = weakHandle();
+        int turnServerDnsRequestId = ++mTurnServerDnsRequest;
+        DnsRequest::getInstance()->wsResolveDNS(mKarereClient.websocketIO, subString.c_str(),
+                                                [wptr, this, subString, turnServerDnsRequestId]
+                                                (int statusDNS, const std::vector<std::string> &ipsv4, const std::vector<std::string> &ipsv6)
+        {
+            if (wptr.deleted() || turnServerDnsRequestId != mTurnServerDnsRequest)
+                return;
+
+            if (!mKarereClient.mDnsCache.hasRecord(TURNSERVER_SHARE))
+            {
+                mKarereClient.mDnsCache.addRecord(TURNSERVER_SHARE, subString);
+            }
+            else if (mKarereClient.mDnsCache.getUrl(TURNSERVER_SHARE).host != subString)
+            {
+                mKarereClient.mDnsCache.removeRecord(TURNSERVER_SHARE);
+                mKarereClient.mDnsCache.addRecord(TURNSERVER_SHARE, subString);
+            }
+
+            if (ipsv4.size() || ipsv6.size())
+            {
+                std::vector<std::string> ips;
+                ips.push_back("127.0.0.1");
+                mKarereClient.mDnsCache.setIp(TURNSERVER_SHARE, ipsv4, ipsv6);
+                KR_LOG_DEBUG("New IP for Turn servers: ipv4 - %s     ipv6 - %s", ipsv4.size() ? ipsv4[0].c_str() : "", ipsv6.size() ? ipsv6[0].c_str() : "");
+                StaticProvider iceServer = getTurnServerProvider(ipsv4[0], ipsv6[0]);
+                setIceServers(iceServer);
```

Comment:
Here, probably it's more appropriate to use the `addIceServer()` instead, so we keep the original list plus the new server, instead of replace it.

Reply:
I use a `setIceServer` instead of `addIceServer` to ensure webrtc use the url returned by the API. If I use `add`, webrtc can use as first option one of the oldest